tests/sentry/hybridcloud/test_organization.py

```diff
@@ -76,23 +79,23 @@ def order_things_by_id(a: list[Any], b: list[Any]) -> None:
     a.sort(key=lambda i: find_ordering(b_ids, i.id))
 
 
-def assert_for_list(a: list[Any], b: list[Any], assertion: Callable) -> None:
+def assert_for_list(a: list[Any], b: list[Any], assertion: Callable[[Any, Any], None]) -> None:
     assert len(a) == len(b)
     order_things_by_id(a, b)
```

Comment:
probably not your issue but I would maybe rewrite this as `a: list[HasId], b: list[HasId]` and instead of `order_things_by_id` and `find_ordering` I would do `.sort(lambda item: item.id)` on both and compare that way.

would probably also want

```python
class HasId(Protocol):
    @property
    def id(self) -> int: ...
```

Reply:
This one turned out be more of a can of worms than expected, going to leave it as-is for now since it's just a test assertion